### PR TITLE
Handle user disconnections

### PR DESCRIPTION
**Pull request by OmarQurashi868**
- Made Socket nullable in Message and `TextMessage` so we can create `TextMessage`s easily
- Implemented `OnClose()` to delete the User from `UserList` and do logging and alert

[Trello ticket](https://trello.com/c/lF4dd3JL/89-handle-user-disconnections)

### CODE REVIEW

File: Backend/ServerModules/ServerBehavior.cs [hunk 1]
```diff
@@ -68,5 +68,24 @@ protected override void OnOpen()
...
+            {
+                CLogger.Error("Disconnected user not found in UserList");
+                return;
+            }
+
+            UserManager.UsersList.Remove(user);
+            CLogger.Event($"User disconnected: {user.Username}");
+
+            TextMessage disconnectAlert = new TextMessage(null, "");
+            disconnectAlert.Content = $"User {user.Username} has disconnected";
```
Comment (ahmed-okkio) [inline Backend/ServerModules/ServerBehavior.cs]: Make a function called SendAlert that accepts  string in LSMPBehavior

File: Backend/ServerModules/ServerBehavior.cs [hunk 2]
```diff
@@ -68,5 +68,24 @@ protected override void OnOpen()
...
+
+            SendToAll(disconnectAlert);
```
Comment (OmarQurashi868) [inline Backend/ServerModules/ServerBehavior.cs]: I should send UserList to all after disconnection

Comment (OmarQurashi868) [reply Backend/ServerModules/ServerBehavior.cs]: Part of the other pr yikes

File: Backend/ServerModules/ServerBehavior.cs (hunk 1 above)
Comment (OmarQurashi868) [reply Backend/ServerModules/ServerBehavior.cs]: Done

**Review by aeltoum — APPROVED** [review]

**Pull request merged**